source/beamline/BioXAS/BioXASSSRLMonochromatorRegionControl.cpp

```diff
@@ -0,0 +1,846 @@
+{
+	if (!brakeStatus_ && brakeStatus_ != brakeStatus) {
+		brakeStatus_ = brakeStatus;
+		addChildControl(brakeStatus_);
+
+		connect( brakeStatus_, SIGNAL(valueChanged(double)), this, SIGNAL(brakeStatusChanged(double)) );
+		connect( brakeStatus_, SIGNAL(movingChanged(bool)), this, SIGNAL(movingChanged(bool)) );
+	}
+}
+
+void BioXASSSRLMonochromatorRegionControl::setBraggControl(AMControl *bragg)
+{
+	if (!bragg_ && bragg_ != bragg) {
+		bragg_ = bragg;
+		addChildControl(bragg_);
+
+		connect( bragg_, SIGNAL(movingChanged(bool)), this, SIGNAL(movingChanged(bool)) );
+	}
+}
+
+void BioXASSSRLMonochromatorRegionControl::setBraggAtCrystalChangePositionStatusControl(AMControl *inPosition)
+{
+	if (!braggAtCrystalChangePositionStatus_ && braggAtCrystalChangePositionStatus_ != inPosition) {
+		braggAtCrystalChangePositionStatus_ = inPosition;
+		addChildControl(braggAtCrystalChangePositionStatus_);
+
+		connect( braggAtCrystalChangePositionStatus_, SIGNAL(valueChanged(double)), this, SIGNAL(braggAtCrystalChangePositionStatusChanged(double)) );
+		connect( braggAtCrystalChangePositionStatus_, SIGNAL(movingChanged(bool)), this, SIGNAL(movingChanged(bool)) );
+	}
+}
+
+void BioXASSSRLMonochromatorRegionControl::setCrystalChangeControl(AMControl *crystalChange)
+{
+	if (!crystalChange_ && crystalChange_ != crystalChange) {
+		crystalChange_ = crystalChange;
+		addChildControl(crystalChange_);
+
+		connect( crystalChange_, SIGNAL(movingChanged(bool)), this, SIGNAL(movingChanged(bool)) );
+	}
+}
+
+void BioXASSSRLMonochromatorRegionControl::setCrystalChangeCWLimitStatusControl(AMControl *limitStatus)
+{
+	if (!crystalChangeCWLimitStatus_ && crystalChangeCWLimitStatus_ != limitStatus) {
+		crystalChangeCWLimitStatus_ = limitStatus;
+		addChildControl(crystalChangeCWLimitStatus_);
+
+		connect( crystalChangeCWLimitStatus_, SIGNAL(movingChanged(bool)), this, SIGNAL(movingChanged(bool)) );
+	}
+}
+
+void BioXASSSRLMonochromatorRegionControl::setCrystalChangeCCWLimitStatusControl(AMControl *limitStatus)
+{
+	if (!crystalChangeCCWLimitStatus_ && crystalChangeCCWLimitStatus_ != limitStatus) {
+		crystalChangeCCWLimitStatus_ = limitStatus;
+		addChildControl(crystalChangeCWLimitStatus_);
+
+		connect( crystalChangeCCWLimitStatus_, SIGNAL(movingChanged(bool)), this, SIGNAL(movingChanged(bool)) );
+	}
+
+}
+
+void BioXASSSRLMonochromatorRegionControl::setRegionAStatusControl(AMControl *regionStatus)
+{
+	if (!regionAStatus_ && regionAStatus_ != regionStatus) {
+		regionAStatus_ = regionStatus;
+		addChildControl(regionAStatus_);
+
+		connect( regionAStatus_, SIGNAL(valueChanged(double)), this, SLOT(onRegionControlValueChanged()) );
+		connect( regionAStatus_, SIGNAL(movingChanged(bool)), this, SIGNAL(movingChanged(bool)) );
+	}
+}
+
+void BioXASSSRLMonochromatorRegionControl::setRegionBStatusControl(AMControl *regionStatus)
+{
+	if (!regionBStatus_ && regionBStatus_ != regionStatus) {
```

Comment:
A general comment about these ```setXYZControl()``` methods: 

I know that these are only called once inside the constructor, but if you were to use these methods in a more general sense (because you needed to change the controls for some reason) you would get some very bizarre behaviour.  For example, you set the new control normally, but you don't remove the old control from the child list and don't disconnect the signals from the old control.  This would lead to weird misfiring of slots and possibly weird behaviour in your status getters like (connected(), canMove(), etc.).

There is also a bit of an ownership issue.  Does this class own the control?  If so, it should handle deleting it.  If it doesn't, it should have a signal that lets others know that it no longer is looking after that control and those other classes can handle deleting/saving/handling it.

Just some things to keep in mind :)